From 5a469d3c8b9d488acb53746b0c25eb81fc98bd6c Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Wed, 23 Oct 2024 14:47:24 +0200
Subject: [PATCH 1293/1695] drivers: stm32_gpio: fix return code if
 apply_rif_config() fails

Report error code instead of TEE_SUCCESS when apply_rif_config() fails.
This patch fixes commit "core: drivers: add RIF support for GPIOS".

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7bfe3e2bf57a5ad10296e51c56cb872bd8731e65 drivers: stm32_gpio: fix return code if apply_rif_config() fails]
Change-Id: I50e572ccda8bf312a9403a4ddfdca2f1f14043cd
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/413572
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/stm32_gpio.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/core/drivers/stm32_gpio.c b/core/drivers/stm32_gpio.c
index 1c680151e..4eb50c1d8 100644
--- a/core/drivers/stm32_gpio.c
+++ b/core/drivers/stm32_gpio.c
@@ -914,8 +914,10 @@ static TEE_Result apply_rif_config(struct stm32_gpio_bank *bank,
 	io_clrsetbits32(bank->base + GPIO_SECR_OFFSET, gpios_mask,
 			bank->rif_cfg->sec_conf[0]);
 
-	if (!bank->is_tdcid)
+	if (!bank->is_tdcid) {
+		res = TEE_SUCCESS;
 		goto out;
+	}
 
 	for (i = 0; i < bank->ngpios; i++) {
 		if (!(BIT(i) & gpios_mask) ||
@@ -983,7 +985,7 @@ static TEE_Result apply_rif_config(struct stm32_gpio_bank *bank,
 out:
 	clk_disable(bank->clock);
 
-	return TEE_SUCCESS;
+	return res;
 }
 
 /* Forward reference to stm32_gpio_set_conf_sec() defined below */
-- 
2.47.2

